daemon/graphdriver/overlay2/overlay.go

```diff
@@ -314,11 +312,10 @@ func (d *Driver) GetMetadata(id string) (map[string]string, error) {
 	return metadata, nil
 }
 
-// Cleanup any state created by overlay which should be cleaned when daemon
-// is being shutdown. For now, we just have to unmount the bind mounted
-// we had created.
+// Cleanup is needed to satisfy the graphdriver.Driver interface.
+// It does nothing since no cleanup beyond the Put method is necessary.
 func (d *Driver) Cleanup() error {
-	return mount.Unmount(d.home)
+	return nil
```

Comment:
Note, in the case of a live-restore upgrade, the previous umount won't be hit.. and after the upgrade to this new codepath, we now won't be cleaning up a private bindmount the previous version of dockerd setup.

Restarting the machine will eventually clean up this bindmount. I'm not sure if this cleanup code should keep unmounting and just ignore errors to handle post-live-restore cases, or whether it's okay to ignore the trouble since it's an edge case and doesn't cause too many ill effects